client/src/featureform/serving.py

```diff
@@ -126,22 +137,88 @@ def _create_channel(self, host, insecure, cert_path):
             return secure_channel(host, cert_path)
 
     def training_set(self, name, variation, include_label_timestamp):
-        return Dataset(self._stub).from_stub(name, variation)
+        return Dataset(self._stub, resource_client=self._resource_client).from_stub(name, variation)
 
     def features(self, features, entities):
+        return FeatureServer(self._stub, self._resource_client, features, entities)
+
+
+class FeatureServer:
+    def __init__(self, stub, resource_client, features, entities):
+        self._stub = stub
+        self._resource_client = resource_client
+        self._features = features
+        self._entities = entities
+        self._feature_list = []
+        if self._all_ready():
+            self._get_features()
+    
+    def __getitem__(self, item):
+        self._feature_list[item]
```

Comment:
If it isn't all ready and I try to use this it'll just throw an indexed error which will be confusing

Reply:
Changed it to check for readiness and throw a value error on exception